src/defaults/runningSession.tsx

```diff
@@ -1,40 +1,105 @@
+import {
+   Kernel, KernelManager
+} from '@jupyterlab/services';
 
-import { IStatusBar } from './../statusBar';
+import {
+    Widget,
+} from '@phosphor/widgets';
 
-import { IconItem } from '../component';
+import {
+    IStatusBar
+} from './../statusBar';
 
-export class RunningSession extends Widget {
-  constructor(src: string) {
-    super();
 
-    this._src = src;
 
-    ReactDOM.render(<IconItem source={this._src} />, this.node);
-  }
 
-  private _src: string;
+export
+namespace RunningComponent {
+    export
+    interface IState {
+        kernelSession: number;
+    }
+    export
+    interface IProps {
+        kernelManager: KernelManager;
+        shellHost: ApplicationShell;
+    }
+}
+
+export
+class RunningComponent extends React.Component<RunningComponent.IProps, RunningComponent.IState> {
+    state = {
+        kernelSession: 0,
+    };
+
+    constructor(props: RunningComponent.IProps) {
+        super(props);
+        this.props.kernelManager.runningChanged.connect(this.updateKernel);
+    }
+
+    componentDidMount() {
+        Kernel.listRunning().then(value => this.setState({kernelSession: value.length}));
+    }
+
+    updateKernel = (kernelManage: KernelManager, kernels: Kernel.IModel[]) =>  {
+        this.setState({kernelSession: kernels.length});
+    }
+
+    handleClick = () => {
+        console.log('hello');
+        this.props.shellHost.expandLeft();
+        this.props.shellHost.activateById('jp-running-sessions');
+    }
+    render() {
+        return(<div onClick = {this.handleClick}>{this.state.kernelSession} Kernels Active</div>);
+    }
+
+}
+
+export
+class RunningSession extends Widget {
+    constructor(opts: AppShell.IOptions) {
+        super();
+        this._manager = new KernelManager();
+        this._host = opts.host;
+    }
+      onBeforeAttach() {
+            ReactDOM.render(<RunningComponent kernelManager = {this._manager} shellHost = {this._host}/>, this.node);
+      }
+
+      private _host: ApplicationShell = null;
+      private _manager: KernelManager;
 }
 
 /*
  * Initialization data for the statusbar extension.
  */
-export const runningSessionItem: JupyterLabPlugin<void> = {
-  id: 'jupyterlab-statusbar/default-items:icon-item',
-  autoStart: true,
-  requires: [IStatusBar],
-  activate: (_app: JupyterLab, statusBar: IStatusBar) => {
-    statusBar.registerStatusItem(
-      'image-1',
-      new RunningSession(
-        'https://emojipedia-us.s3.amazonaws.com/thumbs/120/apple/129/waving-hand-sign_1f44b.png'
-      ),
-      { align: 'right' }
-    );
-  }
+
+export
+const runningSessionItem: JupyterLabPlugin<void> = {
+    id: 'jupyterlab-statusbar/default-items:icon-item',
+    autoStart: true,
+    requires: [IStatusBar],
+    activate: (app: JupyterLab, statusBar: IStatusBar, manager: KernelManager) => {
+        statusBar.registerStatusItem('image', new RunningSession( {host: app.shell} ), {align: 'left'});
+    }
 };
+
+export
+namespace AppShell {
```

Comment:
usually the namespace name should mirror the name of the class. In this case use `RunningSession` as the name of the namespace.